src/Controller/Categorias/CategoriasOperarController.php

```diff
@@ -32,6 +38,18 @@ public function registrarEditarCategoria(CategoriaRepository $categoriasReposito
         $id = $request->request->get("id");
         $nome = $request->request->get("nome");
 
+        if(empty(trim($nome))){
+            $this->addFlash('danger', "O nome da categoria não pode estar vazio!");
+            return $this->redirectToRoute('editarCategoria', ['id' => $id]);
+        }
+
+        $ConsultarCategoria = $categoriasRepository->buscarCategoria($nome);
+
+        if($ConsultarCategoria){
+            $this->addFlash('danger', "A categoria {$nome} já existe!");
+            return $this->redirectToRoute('editarCategoria', ['id' => $id]);
+        }
```

Comment:
$consultarCategoria = $categoriasRepository->buscarCategoria($nome);

    if ($consultarCategoria) {
      $this->addFlash('danger', "A categoria {$nome} já existe!");
            return $this->redirectToRoute('editarCategoria', ['id' => $id]);
ConsultarCategoria foi alterado para consultarCategoria para seguir o padrão camelCase